neo/SmartContract/Native/Tokens/Nep5Token.cs

```diff
@@ -87,15 +87,25 @@ internal protected virtual void Mint(ApplicationEngine engine, UInt160 account,
             engine.SendNotification(Hash, new StackItem[] { "Transfer", StackItem.Null, account.ToArray(), amount });
         }
 
-        internal protected virtual void Burn(ApplicationEngine engine, UInt160 account, BigInteger amount)
+        /// <summary>
+        /// Burn token, when the account balance less than amount, it will burn all the balance of the account
+        /// </summary>
+        /// <param name="engine">The application engine</param>
+        /// <param name="account">The burning account</param>
+        /// <param name="amount">The amount to be burned</param>
+        /// <returns>The actual amount of burned</returns>
+        internal protected virtual BigInteger Burn(ApplicationEngine engine, UInt160 account, BigInteger amount)
         {
             if (amount.Sign < 0) throw new ArgumentOutOfRangeException(nameof(amount));
-            if (amount.IsZero) return;
+            if (amount.IsZero) return amount;
             StorageKey key = CreateAccountKey(account);
             StorageItem storage = engine.Snapshot.Storages.GetAndChange(key);
             TState state = new TState();
             state.FromByteArray(storage.Value);
-            if (state.Balance < amount) throw new InvalidOperationException();
+            if (state.Balance < amount)
+            {
+                amount = state.Balance;
+            }
```

Comment:
I think `state.Balance < amount` is impossible.

Reply:
Agree, it should not happen, I will change this PR.

Reply:
fixed